### ISSUE TITLE
Missing IPC socket when "--chain" is testnet and non-default "--db-path"

### ISSUE THREAD

**Issue by stevenh512** [ethcore/parity#1846]
I tried running Parity (1.3.0 from git) the other night on a 64-bit Ubuntu 14.04 server. I have one other issue preventing me from making the switch, I'll have to wait for Dapple to fix that (Parity and Dapple apparently don't agree on the JSON RPC spec). But in the mean time, I noticed something strange. I was running two instances of Parity, one on homestead and one on morden, each with their own separate data directory. I realize I don't really need to separate their data directories like this, but I'd like to keep certain things (like private keys) separate.

I ran the Homestead node with this command line:
`parity --chain homestead --jsonrpc-interface 127.0.0.1 --jsonrpc-cors "*" --dapps-interface 127.0.0.1 --nat none`

Everything works as expected, there is a `~/.parity/jsonrpc.ipc` file which I'm assuming is the IPC socket. 

I ran the Morden node with this command line:
`parity --chain morden --port 30304 --db-path /home/deploy/.parity-morden --jsonrpc-interface 127.0.0.1 --jsonrpc-port 8645 --dapps-interface 127.0.0.1 --dapps-port 8081 --signer-port 8181 --nat none --geth`

I also tried specifying the IPC path:
`parity --chain morden --port 30304 --db-path /home/deploy/.parity-morden --ipc-path /home/deploy/.parity-morden/jsonrpc.ipc --jsonrpc-interface 127.0.0.1 --jsonrpc-port 8645 --dapps-interface 127.0.0.1 --dapps-port 8081 --signer-port 8181 --nat none --geth`

In both cases, the Morden node seemed to be running but there was no `jsonrpc.ipc` file anywhere to be found under /home/deploy/.parity-morden.

On a non-default chain, with a non-default data directory, how do I make sure there's a (hopefully geth compatible) IPC socket created?

(side note: RPC, signer and dapp interfaces are only listening on 127.0.0.1 and firewalled to prevent external traffic, if I need them externally I SSH tunnel the ports).

**Comment by arkpar** [comment]
`--geth` sets IPC path to the same path geth would place it (somewhere inside `~/.ethereum`) for compatibility with Mist 
The fact that `--ipc-path ` has no effect with `--geth` is a bug that should be fixed.

**Comment by gavofyork** [comment]
it's clearly stated in the `--help` documentation:

```
--geth              ... Overrides the --ipc-path and --ipcpath options ...
```

**Comment by arkpar** [comment]
Does not seem reasonable. It should override only the default instead.

**Comment by stevenh512** [comment]
Thanks, I must have missed that. I tried it again, running only the Morden node with the command line above, and I see that it does place an IPC file at `~/.ethereum/geth.ipc`. I also see that the reason I saw the IPC file under `~/.parity` where I expected it for the Homestead node is because I forgot to include the `--geth` switch for that node. My mistake, no need to change how `--geth` works.

Although it does seem like a better option for my use case (and one I should have considered in the first place) would be to run each node under a separate user account if I want a geth compatible IPC socket. :)
